File: trunk/PolarMath/src/org/macondo/polar/evaluation/spectrum/HF.java
package org.macondo.polar.evaluation.spectrum;

import java.util.List;

import org.macondo.polar.data.Training;
import org.macondo.polar.evaluation.Evaluation;
import org.macondo.polar.util.FFT;
import org.macondo.polar.util.Periodogram;
import org.macondo.polar.util.Square;

public class HF implements Evaluation<Double>{
	public Double evaluate(Training training) {
		List<Periodogram> periodogram = training.evaluate(new FFT());
		
		return new Square(periodogram, 0.15, 0.4).Calculate();
	}
}

File: PolarMathCS/PolarMath/Evaluation/Spectrum/HF.cs
using PolarMath.Data;
using PolarMath.Util;
using System.Collections.Generic;

namespace PolarMath.Evaluation.Spectrum
{
    public class HF : IEvaluation<double>
    {
        public double Evaluate(SessionData training)
        {
            var periodogram = training.Evaluate( new Lomb() );

            return new Square(periodogram, 0.1, 1).Calculate();//time period 2.5 : 6.67
        }
    }
}
